src/BlockTypes/AbstractProductGrid.php

```diff
@@ -261,7 +261,8 @@ protected function get_products() {
 		$is_cacheable = (bool) apply_filters( 'woocommerce_blocks_product_grid_is_cacheable', true, $this->query_args );
 
 		if ( $is_cacheable ) {
-			$query_hash        = md5( wp_json_encode( $this->query_args ) . __CLASS__ );
+			$hash_args         = apply_filters( 'woocommerce_blocks_product_grid_hash_args', $this->query_args );
```

Comment:
Do you think we should move this line above line 261 ( `$is_caceable = ...` ) above so it can benefit from filtered arguments?

And minor nitpick, but `$query_args` and `woocommerce_blocks_product_grid_query_args` might be slightly more fitting.

Reply:
@joshuatf, on our side, I don't see a reason to prevent caching (whatever the language context). I would keep it here because the `$hash_args` is really needed for writing/reading the transient (not before).

Using the naming `$query_args` could be confusing in my opinion. WPML code is hooking into `\WP_Query` but it does not change the query arguments.

Reply:
To be clear, I'm recommending moving this filter earlier so it can be of better use to other devs wanting to extend query args here.  I think that still works for your use case and would allow `query_args` to be modified if necessary by other plugins.

If this does work for you, we could move the filter to query args array inside `parse_query_args()`.

This would of course pass those extra arguments in `$this->query_args` to `WP_Query`, but I don't think that will cause any issues for unrecognized arguments.  @mikejolley can you confirm this?